Gems/AudioSystem/Code/Source/Engine/AudioSystem.cpp

```diff
@@ -192,26 +192,32 @@ namespace Audio
 
         auto startUpdateTime = AZStd::chrono::system_clock::now();        // stamp the start time
 
-        bool handledBlockingRequests = false;
+        bool handleBlockingRequest = false;
 
-        {   // Process a single blocking request, if any, and release the semaphore on the main thread.
+        {   // Process a single blocking request, if any, and release the semaphore the main thread is trying to acquire.
             // This ensures that main thread will become unblocked quickly.
             // If blocking requests were processed, can skip processing of normal requests and skip having
             // the audio thread block through the rest of its update period.
-            AZStd::scoped_lock lock(m_blockingRequestsMutex);
-            if (!m_blockingRequestsQueue.empty())
+
+            AudioRequestVariant blockingRequest;
             {
-                // Blocking request...
-                AudioRequestVariant& request(m_blockingRequestsQueue.front());
-                m_oATL.ProcessRequest(AZStd::move(request));
-                m_blockingRequestsQueue.pop_front();
+                AZStd::scoped_lock lock(m_blockingRequestsMutex);
```

Comment:
The outer block scope is no logner needed.

Reply:
Will update.